src/hooks/useOperatorTrades.ts

```diff
@@ -1,68 +1,44 @@
-import { Trade as TradeMetaData } from '@cowprotocol/cow-sdk'
-import { getTrades, Order, Trade } from 'api/operator'
+import { getTrades, Order, RawTrade, Trade } from 'api/operator'
 import { useCallback, useEffect, useState } from 'react'
 import { useNetworkId } from 'state/network'
 import { Network, UiError } from 'types'
 import { transformTrade } from 'utils'
-
-type Params = {
-  owner?: string
-  orderId?: string
-}
+import { web3 } from 'apps/explorer/api'
 
 type Result = {
   trades: Trade[]
   error?: UiError
   isLoading: boolean
 }
 
-/**
- * Fetches trades for given filters
- * When no filter is given, fetches all trades for current network
- */
-export function useTrades(params: Params): Result {
-  const { owner, orderId } = params
+type TradesTimestamps = { [txHash: string]: number }
 
-  const [isLoading, setIsLoading] = useState(false)
-  const [error, setError] = useState<UiError>()
-  const [trades, setTrades] = useState<Trade[]>([])
+const tradesTimestampsCache: { [blockNumber: number]: number } = {}
 
-  // Here we assume that we are already in the right network
-  // contrary to useOrder hook, where it searches all networks for a given orderId
-  const networkId = useNetworkId()
+async function fetchTradesTimestamps(rawTrades: RawTrade[]): Promise<TradesTimestamps> {
+  const requests = rawTrades.map(({ txHash, blockNumber }) => {
+    const cachedValue = tradesTimestampsCache[blockNumber]
```

Comment:
nitpick: becasue the cache, caches numbers and not Promise<number>, trades executed at the same block won't hit the cache.

I don't think is a big issue, feel free to ignore.

Reply:
Very good point! Actually, I thought about it but didn't implement it for some reason.
Fixed.